src/models/semantic.py

```diff
@@ -700,33 +710,42 @@ def validation_step_log_metrics(self) -> None:
             "val/loss", self.val_loss, on_step=False, on_epoch=True,
             prog_bar=True)
 
-    def on_validation_epoch_end(self) -> None:
-        miou = self.val_cm.miou()
-        oa = self.val_cm.oa()
-        macc = self.val_cm.macc()
+    def on_validation_epoch_end(self):
 
-        # Log metrics
-        self.log("val/miou", miou, prog_bar=True)
-        self.log("val/oa", oa, prog_bar=True)
-        self.log("val/macc", macc, prog_bar=True)
-        for iou, seen, name in zip(*self.val_cm.iou(), self.class_names):
-            if seen:
-                self.log(f"val/iou_{name}", iou, prog_bar=True)
+        if self.trainer.num_devices > 1:
+            epoch_cm = torch.sum(self.all_gather(self.val_cm.confmat), dim=0)
+        else:
+            epoch_cm = self.val_cm
+        # Reset metrics accumulated over the last epoch
```

Comment:
I thin the call to `.reset()` should happen last. You should compute and log your epoch metrics first and then log them. Maybe in DDP it will work because you copied the content of the metrics object from each process (to be verified) when calling `all_gather()` but I fear it won't work in the single-process case. Just to be sure, please move the `.reset()` calls to the end.